site/src/components/home/config/configDialog.jsx

```diff
@@ -49,59 +49,61 @@ const ConfigInput = ({setting, updateData, updateError}) => {
         label: setting.friendlyName,
         id: setting.name,
         onChange: (e) => setValue(e.target.value),
-        value: value
+        value: value,
+        error: error
     };
     useEffect(() => {
         updateError(setting.name, error);
     }, [setting, updateError, error]);
     useEffect(() => {
         if(value !== setting.value) {
-            if(value === '') {
+            if(value === '' && !setting.emptyAllowed) {
                 updateData(setting.name, undefined);
             } else {
                 updateData(setting.name, value);
             }
         }
     }, [value, updateData, setting]);
     const readOnly = !!setting.readOnly;
+    let min;
+    let max;
     switch(setting.type) {
     case settingType.Integer:
+        min = setting.minimumValue ? setting.minimumValue : (-2)**31;
+        max = setting.maximumValue ? setting.maximumValue : 2**31 -1;
```

Comment:
Won't `setting.minimumValue` or `setting.maximumValue` also evaluate to `false` if they are 0 (which can very well be the case)? 

Same question obviously also applies to the `PositiveBigInteger` and `Float` cases.

Reply:
@KeiranHines I had to look it up (my JS is a bit rusty), but 0 is indeed falsy. Which makes that I think we should apply an explicit strict inequality with `undefined` in these conditional expressions.

Reply:
You are correct. That was a mistake. I have corrected it now.